src/components/OptionListContextProvider.tsx

```diff
@@ -71,6 +74,31 @@ function OptionsListContextProvider({reports, children}: OptionsListProviderProp
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [reports]);
 
+    useEffect(() => {
```

Comment:
@TMisiukiewicz can you please add some comment here to explain what this is good for?

Reply:
added comments to each effect for easier understanding 👍